builders.ncl

```diff
@@ -81,195 +70,158 @@ let { NickelDerivation, Derivation, .. } = import "contracts.ncl" in
           |> std.record.map
             (
               fun _n xs =>
-                # Intersperse ':' between environment variable fragments
-                let values = std.record.values xs in
-                let first = std.array.first values in
-                values
-                |> std.array.drop_first
-                |> std.array.fold_left (fun acc value => nix-s%"%{acc}:%{value}"%) first
+                # Intersperse ' ' between environment variable fragments
+                concat_strings_sep " " (std.record.values xs)
```

Comment:
Actually, using `structuredAttrs` mean that we can directly pass a record as `buildInputs` (it'll be passed as an associative array rather than a numerically-indexed array, but that doesn't change anything). So we can get rid of the concat dance altogether

Reply:
> Could we instead use structuredAttrs? stdenv is supposed to be able to handle that just fine now, and that would be nicer than doing this kind of concatenation ourselves

It gets annoying fast. Since `nix-s%"..."%` gets translated to a record (`{ prefix = ..., tag = ..., fragments = ... }`) internally, we have to distinguish between that sort of record and actual records of nix strings. And when I did that, I got `error: boost::bad_format_string: format-string is ill-formed` from Nix :D

Reply:
OK, let's keep away from it for now then.

> I got error: boost::bad_format_string: format-string is ill-formed from Nix :D

:partying_face:

Reply:
Actually, figured it out. I've kept `env` and `structured_env`.